Yoast/wordpress-seo
# Refactor robots helper to use string input and ignore the presentation

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where a fatal error would occur in the robots helper when disallowing search engines from indexing your site.

## Relevant technical choices:

* Due to `generate_robots` calling the filter we can never use `presentation->robots` in that filter. Went for the quick fix to ignore that in the helper now.

## Test instructions

This PR can be tested by following these steps:

* Set your site to not index: Settings -> Reading -> Search Engine Visibility
* Check any frontend page and/or API route: `wp-json/wp/v2/posts`. There should not be a fatal anymore and the robot values should always hold noindex.
  * If `index` before, it should be changed to `noindex`.

## UI changes

## Documentation

## Quality assurance


Fixes #
